articles/api-auth/which-oauth-flow-to-use.md

```diff
@@ -0,0 +1,35 @@
+- **User Agent**: the agent used by the Resource Owner to interact with the Client, for example a browser or a native application.
+:::
+
+## Is the Client the Resource Owner?
+
+The first decision point is about whether the party that requires access to resources is a machine. In this case of machine to machine authorization, the Client is also the Resource Owner. No end-user authorization is needed in this case. An example is a cron job that uses an API to import information to a database. In this example the cron job is the Client and the Resource Owner since it holds the Client Id and Client Secret and uses them to get an access token from the Authorization Server.
+
+## Is the Client a web app executing on the server?
+
+If the Client is a regular web app executing on a server then the **Authorization Code Grant** is the flow you should use. Using this the Client can retrieve an access token and, optionally, a refresh token. It's considered the safest choice since the access token is passed directly to the web server hosting the Client, without going through the user's web browser and risk exposure.
+
+## Is the Client absolutely trusted with user credentials?
+
+This decision point may result to suggesting the **Resource Owner Password Credentials Grant**. In this flow the end-user is asked to fill in credentials (username/password) typically using an interactive form. This information is later on sent to the Client and the Authorization Server. It is therefore imperative that the Client is absolutely trusted with this information.
+
+## Is the Client a native app or a SPA?
+
+If the Client is a Single Page Application (meaning an application running in a browser using a scripting language such as Javascript) then most of the times the [Implicit Grant](/api-auth/grant/implicit) is used. In this case, instead of getting an authorization code that needs to be exchanged for an access token, the Client retrieves directly an access token. On the plus side, this is more efficient since it reduces the number of round trips required to get an access token. However, a security consideration is that the access token is exposed on the client side. Since we are talking about applications running in a browser, viewing the access token is fairly easy. One counter measure is to use [the `state` authentication parameter](/protocols/oauth-state). Also, it should be noted that **Implicit Grant** does not return a refresh token because the browser cannot keep it private. You need to get a new access token, using the `/authorize` endpoint of the [Auth0 Authentication API](/api/authentication), each time the old expires.
+
+**NOTE:** A workaround to improve user experience is to use `prompt=none` when you invoke the `/authorize` endpoint. This will not display the login dialog or the consent dialog so the user is not aware what's happening behind the scenes.
```

Comment:
A minor note here,
```
is not aware what's happening behind the scenes.
```
the customer will see all the redirects happening, so the UI will have to reload. The workaround for this to keep in under the hood, is to call `/authorize` from a hidden `iframe` and extract the new access token from the parent frame.

Maybe what we can say is something down the lines of `while SPAs cannot use refresh tokens, they can take advantage of other mechanics that provide the same function`